Amazon.IonDotnet/Internals/Tree/SystemTreeReader.cs

```diff
@@ -128,11 +128,40 @@ public int GetLobByteSize()
 
         public virtual ISymbolTable GetSymbolTable() => _systemSymbols;
 
-        public IEnumerable<SymbolToken> GetTypeAnnotations()
+        public string[] GetTypeAnnotations()
+        {
+            List<string> annotations = new List<string>();
+            foreach (SymbolToken symbolToken in _current.GetTypeAnnotations())
+            {
+                annotations.Add(symbolToken.Text);
+            }
+
+            return annotations.ToArray();
+        }
+
+        public IEnumerable<SymbolToken> GetTypeAnnotationSymbols()
         {
             return _current.GetTypeAnnotations();
         }
 
+        public bool HasAnnotation(string annotation)
+        {
+            if (annotation == null)
+            {
+                throw new ArgumentNullException(nameof(annotation));
+            }
+
+            foreach (SymbolToken symbolToken in _current.GetTypeAnnotations())
+            {
+                if (annotation.Equals(symbolToken.Text))
+                {
+                    return true;
+                }
+            }
+
+            return false;

```

Comment:
Same comment about throwing `UnknownSymbolException` at this point if any symbol with unknown text has been encountered (assuming that's possible in the Tree context, which is worth double-checking).

Reply:
Fixed.